### DISCUSSION TITLE
Third-Party Integrations with YouTube, GitHub, Twitter, etc.?

### DISCUSSION THREAD

**Discussion by jskherman** [WebCrateApp/webcrate#78] [Ideas] [answered]
Would it be possible to have third-party integrations with WebCrate? For example, adding automatically links of:
- liked videos on YouTube
- starred repos on GitHub
- liked/bookmarked tweets on Twitter

Would scheduled actions be a way to do this? Or webhooks via the API (once finalized)?

**Comment by BetaHuhn** [comment, marked as answer]
Yeah it would definitely be possible via a variety of methods, webhooks might be the easiest. Not sure if it makes sense to add all of these to WebCrate directly since there are so many potential third-party apps/services to integrate with.

I think a better approach would be to make WebCrate extendable on its own and let others build integrations using Space. For example you could already create a [Space micro](https://deta.space/docs/en/basics/micros) that receives a [GitHub webhook request](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#star) when you star something and then talks with your WebCrate instance via [API keys](https://deta.space/docs/en/basics/extending_apps#api-keys) to add the link. 

You could even publish these integrations as your own Space apps so others can use it as well.

**Reply by jskherman** [reply]
I see, thanks for the heads up! By chance, do you have some sort of timeline for when the API is finalized? If there's none, that's OK too.

**Reply by BetaHuhn** [reply]
The HTTP API already is "finalized". The documentation for it is just not publicly listed in the docs and hasn't been fully written. You can use the [partial API docs](https://webcrate.app/docs/api/) and/or check what API requests the app makes if you want to use it yourself.

I am planning on improving the docs for it (hopefully) soon and once Space has more built-in interoperability features extend WebCrate to support them as well to make it easier to integrate with WebCrate without the HTTP API.

**Reply by jskherman** [reply]
Thanks for taking the time to answer! Sorry if I must have misunderstood the writing in the Docs.
